### PR TITLE
Docs: Add documentation on how to debug backend plugins

### PR DESCRIPTION
**Pull request by xnyo**
**What is this feature?**

Documentation for the changes made by https://github.com/grafana/grafana-plugin-sdk-go/pull/621.

Adds instructions on how to debug backend plugins from Visual Studio Code, GoLand or other IDEs.

**Why do we need this feature?**

To make users aware of the new backend debug mode added to the sdk

**Who is this feature for?**

Backend plugin developers

**Which issue(s) does this PR fix?**:

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (andresmgot) [inline docs/sources/developers/plugins/development-with-local-grafana.md:93]: maybe worth mentioning first to  `cd ` in the plugin folder

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (andresmgot) [inline docs/sources/developers/plugins/development-with-local-grafana.md:129]:
```suggestion
> Note: If you re-run the configuration, the plugin will be automatically reloaded in Grafana).
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (andresmgot) [inline docs/sources/developers/plugins/development-with-local-grafana.md:127]:
```suggestion
1. Start Grafana, if it's not already running
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (mckn) [inline docs/sources/developers/plugins/development-with-local-grafana.md:89]:
```suggestion
You can run a backend plugin and attach a debugger to it, which allows you to set breakpoints and debug your backend plugin directly from your IDE of choice.
```
Small nit to make it less repetitive

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (andresmgot) [inline docs/sources/developers/plugins/development-with-local-grafana.md:165]: I think this is `dist/standalone.txt` and `dist/pid.txt`?

**Review by andresmgot — COMMENTED** [review]
Good content! I have some minor comments.

**Review by mckn — APPROVED** [review]
Added one small nit but this looks great!

**Comment by xnyo** [comment]
Thanks for the feedback :smile:  I have addressed the changes

**Review by andresmgot — APPROVED** [review]
LGTM! but maybe wait for a review from @josmperez to get docs team thoughts on this

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:87]:
```suggestion
> Note: The following method only works with a local Grafana instance and currently doesn't work with Docker.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:93]:
```suggestion
1. Go to your plugin's folder.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:96]:
```suggestion
   - If not, update it to the latest version:
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:106]:
```suggestion
Now that your plugin is ready to run, follow the instructions bellow for your IDE of choice.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:110]:
```suggestion
1.  If it's not already present, go to your plugin's folder and place the following file inside `.vscode/launch.json`:
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:129]:
```suggestion
1. Press `F5` to run your plugin in debug mode.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:130]:
```suggestion
1. Start Grafana, if it's not already running.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:132]:
```suggestion
> If you re-run the configuration, Grafana will automatically reload the plugin.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:142]:
```suggestion
1. Run the config (with or without the debugger).
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:144]:
```suggestion
1. Start Grafana, if it's not already running.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:146]:
```suggestion
> If you re-run the configuration, Grafana will automatically reload the plugin.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:152]:
```suggestion
1. Build the executable with debug flags.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:156]:
```suggestion
1. Run the plugin's executable (inside `dist`) with `-standalone -debug` flags.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:160]:
```suggestion
1. Attach a debugger to the process.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:164]:
```suggestion
> If you re-run the plugin executable, Grafana will automatically reload it. 
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:168]:
```suggestion
- All logs are printed in the plugin's `stdout` rather than in Grafana logs.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:169]:
```suggestion
- If the backend plugin doesn't serve requests after turning off debug mode, you can force reset the standalone mode. To do so, delete the files `dist/standalone.txt` and `dist/pid.txt` alongside the executable file, and then restart Grafana.
```

File: docs/sources/developers/plugins/development-with-local-grafana.md
Comment (josmperez) [inline docs/sources/developers/plugins/development-with-local-grafana.md:170]:
```suggestion
- We currently do not support debugging backend plugins running inside Docker.
```

**Review by josmperez — CHANGES_REQUESTED** [review]
Edits for Grafana style guidelines.

**Comment by xnyo** [comment]
Thanks for the review @josmperez ! I have addressed the changes, could you please check again? Thanks :smile:

**Review by josmperez — APPROVED** [review]

**Pull request merged**
